server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "analytics":{
+ *              "jvm": 0.4,
+ *              "mpde": "enforced",
+ *              "_id": "fafjafjkaf9ag8a9ga9g7ag0aagaga"
+ *      }
+ * }
+ */
+@ExperimentalApi
+public class ResourceLimitGroup extends AbstractDiffable<ResourceLimitGroup> implements ToXContentObject {
+
+    public static final int MAX_CHARS_ALLOWED_IN_NAME = 50;
+    private final String name;
+    private final String _id;
+    private final ResourceLimitGroupMode mode;
+    private final Map<String, Object> resourceLimits;
+
+    // list of resources that are allowed to be present in the ResourceLimitGroupSchema
+    public static final List<String> ALLOWED_RESOURCES = List.of("jvm");
+
+    public ResourceLimitGroup(String name, String _id, ResourceLimitGroupMode mode, Map<String, Object> resourceLimits) {
+        Objects.requireNonNull(name, "ResourceLimitGroup.name can't be null");
+        Objects.requireNonNull(resourceLimits, "ResourceLimitGroup.resourceLimits can't be null");
+        Objects.requireNonNull(mode, "ResourceLimitGroup.mode can't be null");
+        Objects.requireNonNull(_id, "ResourceLimitGroup._id can't be null");
+
+        if (name.length() > MAX_CHARS_ALLOWED_IN_NAME) {
+            throw new IllegalArgumentException("ResourceLimitGroup.name shouldn't be more than 50 chars long");
```

Comment:
nit: can use variable here rather than static constant (50) in case we change this in the future

Reply:
Until the var is controlled dynamically in a running process, the var might not add much value. Am I missing something ?